src/objectExplorer/objectExplorerService.ts

```diff
@@ -135,6 +135,7 @@ export class ObjectExplorerService {
                 const credentials = self._sessionIdToConnectionCredentialsMap.get(result.sessionId);
                 const children = result.nodes.map(node => TreeNodeInfo.fromNodeInfo(node, result.sessionId,
                     self._currentNode, credentials));
```

Comment:
You use _currentNode here but your fix seems to indicate that the "currentNode" might not actually be the one you want. In that case isn't this going to result in you getting the children for a different node than the new one you find with the code you added and thus result in the same problem as before?